node/src/components/consensus/era_supervisor.rs

```diff
@@ -55,9 +57,12 @@ use crate::{
     utils::WithDir,
 };
 
-/// The number of recent eras to retain. Eras older than this are dropped from memory.
-// TODO: This needs to be in sync with AUCTION_DELAY/booking_duration_millis. (Already duplicated!)
-const RETAIN_ERAS: u64 = 4;
+/// The unbonding period, in number of eras. After this many eras, a former validator is allowed to
+/// withdraw their stake, so their signature can't be trusted anymore.
+///
+/// A node keeps `2 * BONDED_ERAS` past eras around, because the oldest bonded era could still
+/// receive blocks that refer to `BONDED_ERAS` before that.
+const BONDED_ERAS: u64 = DEFAULT_UNBONDING_DELAY - AUCTION_DELAY;
```

Comment:
The question is which is the oldest era that can still be trusted if we're currently in era `N`.

If I submit an unbonding request in era `U`, my understanding is that in era `U`'s switch block, the auction contract first increments the era counter and then determines the validators for era `U + AUCTION_DELAY + 1`, so that will be the first era where I'm definitely not a validator anymore. I will get my money back at the end of era `U + DEFAULT_UNBONDING_DELAY`, so that will be the last era where era `U + AUCTION_DELAY` can still be trusted. If we're now in era `N = U + DEFAULT_UNBONDING_DELAY`, the oldest trusted era is therefore `N - (DEFAULT_UNBONDING_DELAY - AUCTION_DELAY)`.

(@mpapierski: Correct me if I'm wrong!)

Reply:
> I will get my money back at the end of era `U + DEFAULT_UNBONDING_DELAY`

I see, that's where my understanding differed from yours. I thought that `DEFAULT_UNBONDING_DELAY` counts from `U + AUCTION_DELAY + 1` – i.e. the era where you become validator in.

Reply:
So if I became a validator in era-10, and no one outbid me for the next `DEFAULT_UNBONDING_DELAY` eras, I can withdraw without a delay in era `era-10 + DEFAULT_UNBONDING_DELAY + 2` ?